atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -650,13 +660,158 @@ def set_build_image(self):
+                else:
+                    raise RuntimeError("Build annotation has unknown 'kind' %s" %
+                                       scratch_from['kind'])
+            elif not need_manifest:
+                use_same_image = True
+            else:
+                raise RuntimeError("Build wasn't created from BuildConfig and neither"
+                                   "has 'from' annotation, which is needed for specified arch")
+
+        osbs = get_openshift_session(self.workflow, self.openshift_fallback)
+
+        # inspect BuildConfig
+        if build_config_name:
+            try:
+                build_config = osbs.os.get_build_config(build_config_name)
+            except OsbsException:
+                raise RuntimeError("Build config not found : %s" % build_config_name)
+
+            try:
+                build_name = build_config['spec']['strategy']['customStrategy']['from']['name']
+                build_kind = build_config['spec']['strategy']['customStrategy']['from']['kind']
+            except KeyError:
+                raise RuntimeError("BuildConfig object is malformed")
+
+            if build_kind == 'DockerImage':
+                build_image = build_name
+            elif build_kind == 'ImageStreamTag':
+                imagestream = build_name
+            else:
+                raise RuntimeError("BuildConfig object has unknown 'kind' %s" % build_kind)
+
+        # if imageStream is used
+        if imagestream:
+            try:
+                tag = osbs.get_image_stream_tag(imagestream).json()
+            except OsbsException:
+                raise RuntimeError("ImageStreamTag not found %s" % imagestream)
+
+            try:
+                tag_image = tag['image']['dockerImageReference']
+            except KeyError:
+                raise RuntimeError("ImageStreamTag is malformed %s" % imagestream)
+
+            if '@sha256:' in tag_image:
+                try:
+                    labels = tag['image']['dockerImageMetadata']['Config']['Labels']
+                except KeyError:
+                    raise RuntimeError("Image in imageStreamTag '%s' is missing Labels" %
+                                       imagestream)
+
+                release = labels['release']
+                version = labels['version']
+                docker_tag = "%s-%s" % version, release
+                build_image = tag_image[:tag_image.find('@sha256')] + ':' + docker_tag
+            else:
+                build_image = tag_image
+
+        # we don't have build_image and we need it as we want to build on differnt platforms
+        if not build_image and need_manifest:
+            raise RuntimeError("Failed to obtain buidroot image,"
+                               " which is needed for specified arch")
+
+        # we have build_image with sha, but we need tag to check for manifest list,
+        # as we want to build on different platforms
+        if build_image and '@sha256:' in build_image and need_manifest:
```

Comment:
In the future it's possible the digest will actually be for a manifest list, but right now this is fine.